src/meta.rst

```diff
@@ -71,39 +71,128 @@ A full description of selectors is
+
+.. code-block:: yaml
+
+    vtk_with_osmesa:
+      - False
+      - True
+
+You need to rerender the feedstock after this change.
+
+Currently, ``python, vc, r-base`` will create a matrix of jobs for each supported version. If ``python`` is only a build dependency and not a runtime dependency (eg: build script of the package is written in Python, but the package is not dependent on python), use ``build`` section
+
+Following implies that ``python`` is only a build dependency and no Python matrix will be created.
+
+.. code-block:: yaml
+
+    build:
+      - python
+    host:
+
+
+Following implies that ``python`` is a runtime dependency and a Python matrix for each supported python version will be created.
+
+.. code-block:: yaml
+
+    host:
+      - python
+
+
+
+Compilers
+---------
+
+Although ``conda-build=3`` gives the ability to use Anaconda 5 compilers, they should be used only on Windows as packages built with them are incompatible with conda-forge packages built using the CI compilers. All the packages in conda-forge needs to be rebuilt using the compilers and released simultaneously to avoid package incompatibility.
+
+Following is suggested presently,
+
+.. code-block:: yaml
+
+    requirements:
+      build:
+        - {{ compiler('cxx') }}     # [win]
+        - toolchain                 # [unix]
+
+but in the future you should switch to ``{{`compiler('cxx') }}``. An announcement will be made when this should be done.
+
+Reason for holding out using the Anaconda 5 compilers is that the packages built by them are sometimes incompatible with the packages built with the older compilers in CI platforms conda-forge has been using. All the dependencies of a package should be compiled (This is not true for all cases, but it's better to be cautious) with the new compilers before using the new compiler in a package. This presents a problem in that rebuilding a package will break the dependent packages. Therefore, ``conda-forge`` has decided to rebuild all of the packages and upload them all at once. More details on how this is done will be communicated in the future.
 
 
 Building Against NumPy
 ----------------------
 If you have a package which links\* against ``numpy`` you can build against the oldest possible version of ``numpy`` that is forwards compatible.
-That can be achieved by pinning the build requirements and letting "free" the run requirements.
+That can be achieved by pinning the host requirements and letting "free" the run requirements.
 If you don't want to make things complicated you can use
 
 .. code-block:: yaml
 
-    build:
-      - numpy 1.11.*
+    host:
+      - numpy
     run:
-      - numpy >=1.11
+      - {{ pin_compatible('numpy', min_pin='x.x') }}
```

Comment:
The min pins really don't matter.  You can put it here if you really want to, but I don't think it makes a difference in practice.  If anything, if numpy did anything naughty like the magic number snafu in python 3.6.0-3.6.2, you'd want the more accurate lower bound.

Reply:
Do we still need to add `pin_compatible` in light of PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/46 )?